### PR TITLE
DAOS-8040 control: show vmd backing addresses in storage scan (#6746)

### PR DESCRIPTION
**Pull request by tanabarr**
Consolidate the scan result filter logic, said logic restricts results
to only devices in the request list and will return error if controller
details for any member of the list is missing.

In the case where VMD addresses have been specified in the request
device list, check that details of the backing device(s) for the VMD
are contained in the scan results and return them in place of the
requested logical VMD address. Return error if no backing devices data
exists for a given VMD address.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

**Comment by johannlombardi** [comment]
hm, PRs on release/2.0 are passing ... could you please have a closer look?

**Comment by tanabarr** [comment]
> hm, PRs on release/2.0 are passing ... could you please have a closer look?

hardware small and large failed in post provisioning, restarted test stage

File: src/bio/bio_config.c [hunk 1]
```diff
@@ -256,10 +255,9 @@ read_file(const char *filename, size_t *size)
 	FILE *file = fopen(filename, "r");
 	void *data;
 
-	if (file == NULL) {
+	if (file == NULL)
```
Comment (jolivier23) [inline src/bio/bio_config.c:258]: checkpatch will not complain about this case where you put a comment there.  I already don't like our style choice to remove braces as I think it makes code harder to read and edit, but I really would prefer braces there.

Comment (tanabarr) [reply src/bio/bio_config.c:258]: Do you really want me to re-push to make a change that goes against our coding standards? if we feel strongly then we should change the coding standard.

**Pull request merged**